pkg/api/login.go

```diff
@@ -326,8 +327,20 @@ func (hs *HTTPServer) Logout(c *contextmodel.ReqContext) {
 		}
 	}
 
+	idTokenHint := ""
+	postLogoutRedirect := isPostLogoutRedirectConfigured(setting.SignoutRedirectUrl)
+
 	// Invalidate the OAuth tokens in case the User logged in with OAuth or the last external AuthEntry is an OAuth one
 	if entry, exists, _ := hs.oauthTokenService.HasOAuthEntry(c.Req.Context(), c.SignedInUser); exists {
+		token := hs.oauthTokenService.GetCurrentOAuthToken(c.Req.Context(), c.SignedInUser)
+		if postLogoutRedirect {
+			if token.Valid() {
+				idTokenHint = token.Extra("id_token").(string)
+				hs.log.Debug("post_logout_redirect_uri is configured. id token configured is", "id_token:", idTokenHint)
```

Comment:
I think we should not put the token into console for below reasons:
1. id_token can be a very long string, it can make it hard while searching through logs and log size might increase.
2. If logs are stored somewhere It'll log user information including personal information.
3. I am not 100% sure but the id_token could be used to exchange for access_token. [Ref Link](https://connect2id.com/learn/openid-connect#:~:text=Token%20exchange%20%2D%2D%20The%20ID,authorisation%20server%20(RFC%208693))

Reply:
Sure i would remove the debug log.

Reply:
Paragraph 5 in the provided link references [RFC 8693](https://www.rfc-editor.org/rfc/rfc8693) which defines a flow to exchange the idToken into an accessToken for some service.

The same identiy may be permitted to receive several access tokens for several audiences, etc.

It's like showing your ID card to get a festival ticket that you bought before, and later buying beer with the same ID card.

=> idTokens are credentials and should not be logged. 

By removing the signature and thus only logging the payload of the token the exchange is not possible, but the idToken payload still might contain sensitive personal data.

Reply:
@DanielHons From the token that is available in grafana, fetch id token and then send it to IAM ? Is this the comment?

Reply:
I am seeing same being used ds_proxy.go: director() method. We have validated this behavior with Keycloak. Other IAM's i am not aware.